training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala

```diff
@@ -314,38 +318,63 @@ object AdditiveModelTrainer {
       LinearRankerUtils.makePointwiseFloat(input, config, key)
   }
 
-  private def modelInitialization(input: RDD[Example],
-                                  params: AdditiveTrainerParams): AdditiveModel = {
-    // add functions to additive model
-    val initialModel = if (params.initModelPath == "") {
-      None
-    } else {
-      TrainingUtils.loadScoreModel(params.initModelPath)
-    }
-
+  private def modelInitialization(sc: SparkContext,
+                                  input: RDD[Example],
+                                  params: AdditiveTrainerParams,
+                                  config: Config): AdditiveModel = {
     // sample examples to be used for model initialization
-    val initExamples = input.sample(false, params.subsample)
-    if (initialModel.isDefined) {
-      val newModel = initialModel.get.asInstanceOf[AdditiveModel]
-      initModel(params.minCount, params, initExamples, newModel, false)
-      newModel
-    } else {
+    if (params.initModelPath == "") {
       val newModel = new AdditiveModel()
-      initModel(params.minCount, params, initExamples, newModel, true)
+      initModel(sc, config, params, input, newModel, true)
       setPrior(params.priors, newModel)
       newModel
+    } else {
+      val newModel = TrainingUtils.loadScoreModel(params.initModelPath)
+        .get.asInstanceOf[AdditiveModel]
+      initModel(sc, config, params, input, newModel, false)
+      newModel
     }
   }
 
   // Initializes the model
-  private def initModel(minCount: Int,
+  private def initModel(sc: SparkContext,
+                        config: Config,
                         params: AdditiveTrainerParams,
-                        examples: RDD[Example],
+                        input: RDD[Example],
                         model: AdditiveModel,
                         overwrite: Boolean) = {
+    if (params.dynamicBuckets) {
+      val linearFeatureFamilies = params.linearFeatureFamilies
+      val options = NDTreeBuildOptions(
+        maxTreeDepth = config.getInt("max_tree_depth"),
+        minLeafCount = config.getInt("min_leaf_count"))
```

Comment:
instead of passing in config in `initModel`, how about including these parameters in `AdditiveTrainerParams`